Make wait_for return matched groups defined in search_regex. Closes #…

bjorndown opened the pull request:
…25020.

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Fixes #25020

Add module return values `match_groups` and `match_groupdict` that contain any matched groups defined in `search_regex`.
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
wait_for
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->


##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->

Commit:
Make wait_for return matched groups defined in search_regex. Closes #25020.

Commit:
Fix formatting issues.

mattclay commented on test/integration/targets/wait_for/tasks/main.yml:
```diff
@@ -62,13 +62,36 @@
+    - waitfor.elapsed <= 15
+
+- name: setup write keyword to file after 10s
+  shell: rm -f /tmp/wait_for_keyword && sleep 10 && echo "completed data 123" > /tmp/wait_for_keyword
```
Use `{{ output_dir }}/` as a scratch directory instead of `/tmp/`.

The `rm` isn't needed when using `output_dir` since it is empty at the start of each test run.

If possible, use a lower sleep time. I realize 10 seconds was probably copied from the other tests, but as we add more tests those add up. If the test works reliably with 3 or 5 seconds, use that instead.

bjorndown replied:
Changed to use output_dir. Also changed sleeps from 10s to 3s.

mattclay commented on test/integration/targets/wait_for/tasks/main.yml:
```diff
@@ -62,13 +62,36 @@
+
+- name: test wait for keyword in file with match groups
+  wait_for:
+    path: /tmp/wait_for_keyword
```
Use `{{ output_dir }}/` as a scratch directory instead of `/tmp/`.

bjorndown replied:
Done

mattclay commented on test/integration/targets/wait_for/tasks/main.yml:
```diff
@@ -62,13 +62,36 @@
+    - waitfor is successful
+    - "waitfor.search_regex == 'completed'"
+    - waitfor.elapsed >= 5
+    - waitfor.elapsed <= 15
```
The original indentation on these 4 lines is OK and doesn't need to be changed.

bjorndown replied:
Reverted

Commit:
Fix issues raised in review.

- Use output_dir instead of hardcoded /tmp for temp files
- Sleep for only 3s instead of 10s
- Revert indent change

dagwieers reviewed (approved):
The only thing I noticed from the tests was that in most cases it was written as `{{ output_dir }}` but in some cases `{{ output_dir}}` or `{{output_dir}}`. However since it's in the integration tests, I don't think it's a real problem.

abadger commented on test/integration/targets/wait_for/tasks/main.yml:
```diff
@@ -98,7 +121,7 @@
 
 - name: setup start SimpleHTTPServer
-  shell: sleep 10 && cd {{ files_dir }} && {{ ansible_python.executable }} {{ output_dir}}/testserver.py {{ http_port }}
+  shell: sleep 3 && cd {{ files_dir }} && {{ ansible_python.executable }} {{ output_dir}}/testserver.py {{ http_port }}
```
Why change sleep here?  It seems like that might have been tuned for timing in CI or similar.  If it's just to take less time maybe we should change it in a separate PR.

abadger replied:
And the same comment for all the other timing changes.
